packages/editor/src/components/autosave-monitor/index.js

```diff
@@ -7,14 +7,26 @@ import { withSelect, withDispatch } from '@wordpress/data';
 
 export class AutosaveMonitor extends Component {
 	componentDidUpdate( prevProps ) {
-		const { isDirty, editsReference, isAutosaveable } = this.props;
+		const { isDirty, editsReference, isAutosaveable, isAutosaving } = this.props;
+
+		if ( editsReference !== prevProps.editsReference ) {
+			this.didAutosaveForEditsReference = false;
+		}
+
+		if ( ! isAutosaving && prevProps.isAutosaving ) {
+			this.didAutosaveForEditsReference = true;
+		}
```

Comment:
It'd probably useful at this point to include inline comments throughout this entire function that explain the idiosyncrasies of these variables.

Reply:
Yep, that would be sensible.

Reply:
Added in rebased fdd44b2.